v1/message_test.go

```diff
@@ -49,6 +51,41 @@ func TestEncodeMessage(t *testing.T) {
 	require.NoError(t, err)
 	require.EqualValues(t, StatusMessage{
 		Message: testMessageStruct,
+		ID:      crypto.Keccak256(data),
+	}, val)
+}
+
+func TestWrappedMessageWithSignature(t *testing.T) {
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	data, err := EncodeMessage(testMessageStruct)
+	require.NoError(t, err)
+	wrappedMessage, err := WrapMessageV1(data, key)
+	require.NoError(t, err)
+	// Decode it back to a struct because, for example, map encoding is non-deterministic
+	// and it is not possible to compare bytes.
+	val, err := DecodeMessage(wrappedMessage)
```

Comment:
Hm. Looking at it from lib user perspective, we have separate `EncodeMessage` and `WrapMessageV1` but only `DecodeMessage`. Maybe we should have `EncodeMessage(WrapMessageV1(data))` and its counterpart `DecodeMessage(UnwrapMessageV1(data))`.

Reply:
I feel a single DecodeMessage should be ok, as we are always fine decoding V1 messages, while sending we need to make a choice (also by the time we go to v1 we won't have `WrapMessageV1` but only `EncodeMessage` and `DecodeMessage`), if you don't feel strongly about it I would leave it as it for now

Reply:
Ok, that makes sense. We can clean it up later after V1.